pandas/util/testing.py

```diff
@@ -961,7 +961,7 @@ def raise_assert_detail(obj, message, left, right):
 
 def assert_numpy_array_equal(left, right, strict_nan=False,
                              check_dtype=True, err_msg=None,
-                             obj='numpy array'):
+                             obj='numpy array', check_same=None):
```

Comment:
I preferred your suggestion to merge the "copy/same" comparison here because this is actually also used in the ``MultiIndex`` test (which I will include in the ``MultiIndex``-specific PR).